code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -181,24 +181,30 @@
 	R.module = RM
 	R.update_module_innate()
 	RM.rebuild_modules()
+	R.cut_overlays()
 
 	if(RM.use_old_animation && RM.old_animation_length)
 		R.icon = 'icons/hippie/mob/robot_transformations.dmi'
 		R.icon_state = RM.cyborg_base_icon
-		world << "[R.icon_state]"
 		R.dir = SOUTH
 		R.notransform = TRUE
 		flick(R.icon_state, R)
-		sleep(RM.old_animation_length+1)
+		//sleep(RM.old_animation_length+1)
+		addtimer(CALLBACK(src, .proc/transform_to_finale, R), RM.old_animation_length + 1)
 		R.notransform = FALSE

```

Comment:
you dont need this if you have an addtimer, because notransform is set in the transform_to_finale proc right?